### ISSUE TITLE
wsgiref raising incorrect exception description

### ISSUE THREAD

**Issue by betson-fernando** [python/cpython#110001]
# Bug report

### Bug description:

```python
# Add a code block here, if required
```

### CPython versions tested on:

3.11

### Operating systems tested on:

Windows

**Comment by betson-fernando** [comment]
It was doing some tests in make_server function from wsgiref.simple_server.py module by creating an application. 

This is my code:
```python
from wsgiref.simple_server import make_server
import sys

sys.path.append(r"C:\somepath")

#Function that I'm doing some tests
def application (environ, start_response):

    response_body = [
        '%s: %s' % (key, value) for key, value in sorted(environ.items())
    ]
    response_body = '\n'.join(response_body)

    status = '200 OK'
    response_headers = [
        ('Content-Type', 'text/plain'),
        ('Content-Length', str(len(response_body)))
    ]
    start_response(status, response_headers)

    return bytes(response_body, 'utf-8')

# Instantiate the server
httpd = make_server (
    'localhost', # The host name
    8051, # A port number where to wait for the request
    application # The application object name, in this case a function
)

# Wait for a single request, serve it and quit
httpd.handle_request()
```

I know that is strongly recommended that the return from application function must be a list of bytestrings, not a bytestring for better performance. But if the return is a bytestring (not a list of bytestrings), an error is raised as " write() argument must be a bytes instance" in the write() method from BaseHandler class, inside wsgiref.handlers.py module. This is the complete traceback:

```
Traceback (most recent call last):
  File "C:\Users\GEPH-IC\AppData\Local\Programs\Python\Python311\Lib\wsgiref\handlers.py", line 138, in run
    self.finish_response()
  File "C:\Users\GEPH-IC\AppData\Local\Programs\Python\Python311\Lib\wsgiref\handlers.py", line 184, in finish_response
    self.write(data)
  File "C:\Users\GEPH-IC\AppData\Local\Programs\Python\Python311\Lib\wsgiref\handlers.py", line 278, in write
    assert type(data) is bytes, \
           ^^^^^^^^^^^^^^^^^^^
AssertionError: write() argument must be a bytes instance
```

Analyzing the handlers.py module I saw that finish_response() method from BaseHandler class states that application return ('data' variable) must be a list:

```python
def finish_response(self):
        """Send any iterable data, then close self and the iterable

        Subclasses intended for use in asynchronous servers will
        want to redefine this method, such that it sets up callbacks
        in the event loop to iterate over the data, and to call
        'self.close()' once the response is finished.
        """
        try:
            if not self.result_is_file() or not self.sendfile():
                for data in self.result:
                    self.write(data)
                self.finish_content()
        except:
            # Call close() on the iterable returned by the WSGI application
            # in case of an exception.
            if hasattr(self.result, 'close'):
                self.result.close()
            raise
        else:
            # We only call close() when no exception is raised, because it
            # will set status, result, headers, and environ fields to None.
            # See bpo-29183 for more details.
            self.close()
```

If data is a bytestring (not a list of bytestrings), the loop above will iterate over the bytes, returning integers. But data is passed as argument to write() method from the BaseHandler class. And, inside this method:

```python
def write(self, data):
        """'write()' callable as specified by PEP 3333"""

        assert type(data) is bytes, \
            "write() argument must be a bytes instance"

        [...]
```

So that, if data argument is a integer, it will raise that exception.
So, in summary: if data is a bytestring, the for loop will iterate over the bytes, passing integers as argument to write() function, that will raise an AssertionError "write() argument must be a bytes instance".

Because I don't know what is better solution, or forcing a bytestring be a list, or accepting bytestring too, I'll only put a light on this issue.

I expect that it was helpful.
